### DISCUSSION TITLE
trim:edges=south not working as expected

### DISCUSSION THREAD

**Discussion by chapas** [ImageMagick/ImageMagick#3172] [Help]
Hello,
I'm using the following command:

`magick.exe convert a.jpg -fuzz 80% -define trim:edges=south -trim +repage a.jpg`

Expecting only to see the bottom of the image getting trimmed, but top(north) also is getting trimmed.

Can someone advise if I'm doing something wrong.

Regards,
chapas

**Comment by snibgo** [comment]
What version of IM? That seems to be a new feature, perhaps not supported by the version you use.

And: Don't use "magick convert" because that gives you old syntax. Use just "magick" instead.

**Reply by chapas** [reply]
Thanks for your reply:

Version: ImageMagick 7.0.10-60 Q16 x64 2021-01-25 https://imagemagick.org

I have changed "magick convert" to "magick", seem to trim only south.

When trimming borders, can I specify a colour range to be considered, from #DBDBDB to #FFFFFF for instance, meaning: it would only trim if all values in the line/column to trim were inside these.

**Reply by fmw42** [reply]
No color range. But you can specify a color and tolerance on each side in the form of -fuzz XX%.  See https://legacy.imagemagick.org/Usage/color_basics/#fuzz. https://imagemagick.org/script/command-line-options.php#trim  See where it says:  "This option (-trim) removes any edges that are exactly the same color as the corner pixels. Use -fuzz to make -trim remove edges that are nearly the same color as the corner pixels."

**Reply by chapas** [reply]
Won't work for me, for I have multiple files, I only want to trim 'tones' of white in all side, I have 1000 pages with yellow on top and 1000 with white and then yellow on top, yellow's would also get trimmed out on the first 1000 pages.
I don't really wish to select 1000 pages by hand.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/download/beta/ by sometime tomorrow.
